scripts/generate_secrets.py

```diff
@@ -11,23 +16,73 @@ def get_file_contents(folder, filename, trim=False):
     return data
 
 
+def to_bytes(bytes_or_str):
+    if isinstance(bytes_or_str, str):
+        value = bytes_or_str.encode()
+    else:
+        value = bytes_or_str
+    return value
+
+
+def generate_kid_from_key(pem_data):
+    key_as_bytes = to_bytes(pem_data)
```

Comment:
Not sure this is correct, I'd expect to load the PEM data using python cryptography and convert the public key to bytes, otherwise you're hashing the PEM text not the key bytes itself (see https://www.ietf.org/rfc/rfc3280.txt section 4.2.1.2)